#! /usr/bin/env python

PACKAGE='youbot_arm_joints_trajectory'
try:
    import imp
    imp.find_module(PACKAGE)
    from dynamic_reconfigure.parameter_generator_catkin import *
except:
    import roslib; roslib.load_manifest(PACKAGE)
    from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("timeout",   double_t, 0, "timeout", 20, 0, 120)
gen.add("joint0",    double_t, 0, "joint0", 2.93, 0.0, 5.85)
gen.add("joint1",    double_t, 0, "joint1", 1.31, 0.0, 2.7)
gen.add("joint2",    double_t, 0, "joint2", -2.51, -6.0, -0.0)
gen.add("joint3",    double_t, 0, "joint3", 1.73, 0.0, 3.6)
gen.add("joint4",    double_t, 0, "joint4", 2.88, 0.0, 5.7)

gen.add("record", bool_t, 0, "record trajectory", False)
gen.add("clear", bool_t, 0, "clear trajectory", False)
gen.add("run",    bool_t, 0, "run trajectory", False)
gen.add("load",    bool_t, 0, "load trajectory", False)
gen.add("id",    int_t, 0, "trajectory id", 0,0,20)
gen.add("traj_name",  str_t,    0, "A string parameter",  "NULL")
exit(gen.generate(PACKAGE, "youbot_arm_joints_trajectory", "Controller"))
